util/chplenv/chpl_llvm.py

```diff
@@ -745,6 +746,7 @@ def filter_llvm_config_flags(flags):
             flag.startswith('-O') or
             flag == '-pedantic' or
             flag == '-Wno-class-memaccess' or
+            (gnu and flag.startswith('-stdlib=')) or
```

Comment:
libc++ is the default C++ library on Mac OS X, so it would make more sense to me if this also checked that the platform is Mac OS X in addition to what you have here.

Reply:
`-stdlib` is not a valid flag in any GCC version, on any platform; thus, I believe it's safe to strip it out here (stack [overflow answer as evidence](https://stackoverflow.com/a/50407611))

Reply:
That S.O. question says that newer GCC supports `-stdlib` on some platforms / some configurations. I still think it'd be better to limit this to Mac OS X, since that is the only platform we know of where this could come up. However the main thing is that if we guess wrong about where to draw this line, we want as understandable an error as possible. I am expecting that `-stdlib` being thrown when it is not supported is more understandable than say a screen full of linker errors.